_notebooks/textbook.py

```diff
@@ -130,3 +144,22 @@ def clean_raw_text(self, disable_pysbd=False):
             if not disable_pysbd:
                 clean_text = "\n".join(seg.segment(clean_text))
             chapter.clean_text = clean_text
+    
+    def resolve_coreference(self):
+        """Uses spacy pipleline as a base and extends it
+        using neuralcoreference to process the coreferences
+        in the plain text that you get.
+
+        Saving both resolved text as well as the coreference
+        clusters in the chapter.
+        """
+        for chapter in tqdm(self.chapters):
+            if not chapter.clean_text :
+                print(
```

Comment:
This is where you can raise an Exception instead of a silent failure - which is what a `return` statement does. 

See: https://docs.python.org/3/tutorial/errors.html#raising-exceptions